adr/ADR-34.md

```diff
@@ -0,0 +1,55 @@
+# Title JetStream Consumers Multiple Filters
+
+| Metadata | Value                     |
+|----------|---------------------------|
+| Date     | 2023-01-18                |
+| Author   | @Jarema                   |
+| Status   | Approved                  |
+| Tags     | jetstream, client, server |
+
+## Context and Problem Statement
+
+Initially, JetStream Consumers could have only one Filter Subject.
+As the number of feature requests to specify multiple subjects increased, this feature was added.
+That could also reduce the number of Consumers in general.
+
+## Context 
+
+Server PR: https://github.com/nats-io/nats-server/pull/3500
+
+## Design
+
+### Client side considerations
+
+To implement the feature without any breaking changes, a new field should be added to Consumer config, both on the server and the clients.
+
+1. The new field is:  
+`FilterSubjects []string json:"filter_subjects"`  
+
+2. Subjects can't overlap each other and have to fit the interest of the Stream.
+In case of overlapping subjects, error (10136) will be returned.
+
+3. Only one of `FilterSubject` or `FilterSubjects` can be passed. Passing both results in an error (10134)
+
+4. Until future improvements, only old JS API for consumer creation can be used (the one without `FilterSubject`) in consumer create request. Using new API will yield an error (10135).
+
```

Comment:
`$JS..CONSUMER.CREATE.>`

Reply:
More importantly, if it is an update, the consumer could have been created using the new API, so it could have a `name`, depending on how the client is handling this it may reject the `name`. Effectively, if `Array.isArray(config.filter_subjects)`, it should use the old API. Server will properly catch if `filter_subject` and filter_subjects` are both set.

Reply:
Do we prevent a consumer update from switching between FilterSubjet and FilterSubjects?

Reply:
No.
User can switch the consumer from being single-filtered to multi-filtered.
Considering above, we might want to deny it until we land solution for multiple filters and new API.